src/commands/newFile.ts

```diff
@@ -0,0 +1,132 @@
+
+    vscode.commands.executeCommand("vscode-objectscript.explorer.refresh");
+    const uri = DocumentContentProvider.getUri(response.result.name, node.workspaceFolder, node.namespace);
+    if (response.result.ext && response.result.ext[0] && response.result.ext[1]) {
+      // ext is an array containing the responses for actions 1 and 7
+      fireOtherStudioAction(OtherStudioAction.CreatedNewDocument, uri, response.result.ext[0]);
+      fireOtherStudioAction(OtherStudioAction.FirstTimeDocumentSave, uri, response.result.ext[1]);
+    }
+    vscode.window.showTextDocument(uri, { preview: false });
+  } catch (error) {
+    vscode.window.showErrorMessage(error.message);
+  }
+}
+
+function getInputBoxOptions(node: NodeBase): vscode.InputBoxOptions {
+  let fileExtension: string;
+  let filePath: string;
+
+  if (node instanceof PackageNode || node instanceof RootNode) {
+    if (node.category === "CSP") {
+      filePath = node.fullName ? node.fullName + "/" : "";
+    } else {
+      switch (node.category) {
+        case "RTN":
+          fileExtension = "mac";
+          break;
+        case "OTH":
+          fileExtension = "";
+          break;
+        default:
+          fileExtension = node.category.toLowerCase();
+      }
+      filePath = node instanceof PackageNode ? node.fullName + "." : "";
+    }
+  } else if (node instanceof ClassNode || node instanceof RoutineNode) {
+    const clickedFileExtension = node.fullName.match(/.*\.(.*)/)[1];
+    // CSP files
+    if (node.fullName.includes("/")) {
+      filePath = node.fullName.match(/(.*\/)/)[1];
+    }
+    // Classes and Routines
+    else if (clickedFileExtension.match(/(mac|inc|cls)/i)) {
+      const match = node.fullName.match(/(.*\.).*\./);
+      filePath = match ? match[1] : "";
+      fileExtension = clickedFileExtension;
+    }
+    // Other
+    else {
+      filePath = "";
+    }
+  }
+
+  let prompt: string;
+  let sampleName: string;
+  switch (fileExtension) {
+    case "cls":
+      prompt = "Create a class:";
+      sampleName = "ClassName";
+      break;
+    case "mac":
+    case "inc":
+      prompt = "Create a routine:";
+      sampleName = "RoutineName";
+      break;
+    default:
+      prompt = "Create a file:";
+      sampleName = "FileName";
+  }
+
+  return {
+    prompt: prompt,
+    value: filePath + sampleName + (fileExtension ? "." + fileExtension : ""),
+    valueSelection: [filePath.length, filePath.length + sampleName.length],
+  };
+}
+
+function getContent(fileName: string): string[] {
+  const [name, fileExtension] = fileName.match(/(.*)\.(.*)/).slice(1);
+  let contentArray: string[];
+  switch (fileExtension.toUpperCase()) {
+    case "CLS":
+      contentArray = ["Class " + name, "{", "", "}"];
+      break;
+    case "MAC":
+      contentArray = ["ROUTINE " + name, ""];
+      break;
+    case "INC":
+      contentArray = ["ROUTINE " + name + " [Type=INC]", ""];
+      break;
+    case "CSP":
+      contentArray = [
```

Comment:
Note: the file is not necessarily HTML. What happens if you create js/css/etc.?

Reply:
That is a switch on the file extension, so js and css files are empty when created.

Reply:
Oh. Duh. Thanks!

Reply:
@timleavitt are we agreed that the ObjectScript Explorer view will stay read-only? I also like the simplicity of keeping all editing to the normal Explorer view.

Reply:
@rajrsingh yes; I actually wasn't aware of isfs. (Or rather, I'd heard of it and incorrectly equated it to the ObjectScript explorer.) Do we have doc on setting that up somewhere? I did see it in https://community.intersystems.com/post/your-opinion-matters-intersystems-development-environments#comment-82141

Reply:
@timleavitt the best doc we have on it right now is https://github.com/intersystems-community/vscode-objectscript/wiki/Server-side-editing but of course our internal team will be working hard this month on that. They could probably benefit from your usage experience.